Add an e2e test job for federation release-1.4.

madhusudancs opened the pull request:
Also add a build job as a trigger for the test job.

Note for reviewers:
I have already done these things:
  1. Setup a new project - "k8s-jkns-e2e-gce-f8n-1-4"
  2. Create release buckets.
  3. Setup a new domain name for the tests - "k8s-federation-1-4.com."

cc @pwittrock @quinton-hoole @colhom @kubernetes/sig-cluster-federation

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/kubernetes/test-infra/558)
<!-- Reviewable:end -->

nikhiljindal commented on jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-gce.yaml:
```diff
@@ -595,6 +595,26 @@
+                export KUBE_GCE_ZONE="us-central1-f" #TODO(colhom): This should be generalized out to plural case
+                export KUBE_GCS_RELEASE_BUCKET=kubernetes-federation-release
+                export KUBE_GCS_DEV_RELEASE_BUCKET=kubernetes-federation-release
+                export KUBE_NODE_OS_DISTRIBUTION="debian"
```
Is this required?
We should be fine not overwriting this and getting the default value?

nikhiljindal commented on jenkins/job-configs/kubernetes-jenkins/kubernetes-build.yaml:
```diff
@@ -85,5 +85,15 @@
+              export FEDERATION_PUSH_REPO_BASE="gcr.io/k8s-jkns-e2e-gce-f8n-1-4"
+              export KUBE_FASTBUILD=true
+              export KUBE_GCS_RELEASE_BUCKET=kubernetes-federation-release-1-4
+              export KUBE_GCS_RELEASE_BUCKET_MIRROR=kubernetes-federation-release-1-4
```
What is the mirror used for?

madhusudancs commented:
@nikhiljindal addressed your comment. PTAL.

---

Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[jenkins/job-configs/kubernetes-jenkins/kubernetes-build.yaml, line 97 at r1](https://reviewable.io:443/reviews/kubernetes/test-infra/558#-KRVgDGbxDsDhIp1t7AW-r1-97:-KRWPM0FRRvgKK9K0o9a:b1qdiy9) ([raw file](https://github.com/kubernetes/test-infra/blob/3e52c9b8a5d34c315bb8850a4b1ab13d2f82196b/jenkins/job-configs/kubernetes-jenkins/kubernetes-build.yaml#L97)):*
<details><summary><i>Previously, nikhiljindal (Nikhil Jindal) wrote…</i></summary>
> What is the mirror used for?
</details>
No idea. But we don't have time to experiment with this (and possibly waste Phil's and other's time) given the release is next week. I am taking a known working config and just replicating it.

---

*[jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-gce.yaml, line 617 at r1](https://reviewable.io:443/reviews/kubernetes/test-infra/558#-KRVgDGbxDsDhIp1t7AX-r1-617:-KRWPLM6EM7I7K1QvtDK:b-896fix) ([raw file](https://github.com/kubernetes/test-infra/blob/3e52c9b8a5d34c315bb8850a4b1ab13d2f82196b/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-gce.yaml#L617)):*
<details><summary><i>Previously, nikhiljindal (Nikhil Jindal) wrote…</i></summary>
> Is this required?
> 
> We should be fine not overwriting this and getting the default value?
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/558#-:-KRWPdSx1Pr6bMs74E_m:b-f9vnss)*
<!-- Sent from Reviewable.io -->

quinton-hoole commented:
LGTM.  Thanks @nikhiljindal .  I'll merge this manually, as I understand that there is no merge bot running against this repo.

quinton-hoole commented:
Sorry, thanks @madhusudancs, of course.  Too many concurrent PR reviews :)

madhusudancs commented:
@quinton-hoole Thanks for merging this PR!

> Sorry, thanks @madhusudancs, of course. Too many concurrent PR reviews :)

Indeed! :laughing:
